### PR TITLE
Add com.snowplowanalytics.mobile/application/jsonschema/1-0-0

### PR DESCRIPTION
**Pull request by keanerobinson**

### CODE REVIEW

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		"version": "1-0-0"
+	},
+	"type": "object",
+	"properties": {
+		"version": {
+			"type": "string",
+			"description": "Version of the application e.g 1.1.0",
+			"maxLength": 64
+		},
+		"build": {
```
Comment (chuwy) [inline schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: I'm sorry, I'm not really aware of context (pun!) of this task, but I don't think that any application has a `build`, e.g. I cannot think of any build ids in my CLI nor Server apps, so I assume this is optional. Also description of `build` is not very useful.

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		"name": "application",
+		"format": "jsonschema",
+		"version": "1-0-0"
+	},
+	"type": "object",
+	"properties": {
+		"version": {
+			"type": "string",
+			"description": "Version of the application e.g 1.1.0",
+			"maxLength": 64
```
Comment (chuwy) [inline schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: I think 16 or 32 should be enough for longest types of versioning.

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+	"$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+	"description": "Schema for an application context",
```
Comment (chuwy) [inline schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: Also, not very descriptive. Is it something to uniquely identify an application? What kind of application? Mobile? Desktop? Server?

**Comment by chuwy** [comment]
Also @keanerobinson, could you please edit the commit message to conform https://github.com/snowplow/iglu-central/issues/136

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 (hunk 2 above)
Comment (keanerobinson) [reply schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: will amend

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 (hunk 3 above)
Comment (keanerobinson) [reply schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: will amend

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 (hunk 1 above)
Comment (chuwy) [reply schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: Sorry, I actually just noticed `mobile` in vendor, which both answers some of my questions and slightly confuses me. I think something helping to uniquely identify an application is very useful across all platforms, not just mobile. Beside of current tasks, is there any reasons to restrict it to be mobile-only?

**Comment by keanerobinson** [comment]
hey @alexanderdean the schema is intentionally minimal as the plan is to use it solely for automated build / version tracking for users of our iOS / Android SDK's. Myself and @mhadam didn't find any other properties worth tracking out-of-the-box across iOS / Android. 

Furthermore, in most conversations about this with our mobile customers application build / version are the only fields that have been requested (Lifecake / In Crowd Sports / Hudl / Hush).

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 (hunk 1 above)
Comment (keanerobinson) [reply schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: @chuwy apologies I probably should've given more context in my email earlier.

A big focus for the solutions engineering team is the Mobile Foundation (part of the Services OKR's). The whole idea behind Mobile Foundation is to decrease the time to value for our iOS / Android SDK users. The approach as a result is user-led. One decision that we made was to start using a separate vendor for any schema releases as it's a better user experience.

Fully agree that tracking app versions applies not just to Mobile apps but the "Mobile Foundation" release of the application context will be used just for our users of the iOS / Android trackers. @mhadam has already worked on functionality for the iOS/Android trackers to start tracking this automatically which is why we've written a dedicated schema.

**Comment by alexanderdean** [comment]
Makes sense, thanks

File: schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0 [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	"properties": {
+		"version": {
+			"type": "string",
+			"description": "Version number of the application e.g 1.1.0",
+			"maxLength": 32
+		},
+		"build": {
+			"type": "string",
+			"description": "Build name of the application e.g s9f2k2d or 1.1.0 beta",
+			"maxLength": 32
```
Comment (alexanderdean) [inline schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0]: I would make these fields longer. All it takes is for somebody to be using SHA1 hashes or something in their version numbers and we will be in trouble. There's no upside to setting an arbitrarily low limit.

**Comment by keanerobinson** [comment]
Closing this in favour of the [original PR](https://github.com/snowplow/iglu-central/pull/843) which I've branched off and resolved the issues for.

**Pull request closed without merging**